components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/backchain/TransactionBackchainResolutionFlow.kt

```diff
@@ -55,8 +56,12 @@ class TransactionBackchainResolutionFlowVersionedFlowFactory(
 
     override fun create(version: Int, session: FlowSession): SubFlow<Unit> {
         return when {
-            version in 1..50099 -> TransactionBackchainResolutionFlowV1(initialTransactionIds, session)
-            version >= 50100 -> TransactionBackchainResolutionFlowV2(initialTransactionIds, session)
+            version in 1 until CORDA_5_1.platformVersion -> TransactionBackchainResolutionFlowV1(
```

Comment:
Not sure that is the most efficient way of writing this - doesn't this create a huge array of numbers and then check if your version is in that array? Or is there a kotlin optimisation under the hood? Wouldn't this be more efficient:
```suggestion
            version < CORDA_5_1.platformVersion -> TransactionBackchainResolutionFlowV1(
```

Reply:
`until` uses `IntRange`, which is smart enough to store only the ends:
![image](https://github.com/corda/corda-runtime-os/assets/3994645/09c0933e-2c9b-4ce5-9b11-b8190cc801c4)